app.py

```diff
@@ -226,6 +228,17 @@ def _add_constraints(self):
                     if bat.s_goal[t] > 0:
                         self.problem += (self.variables['s'][i][t] >= bat.s_goal[t])
 
+            # Constraint: Minimum battery charge demand (for t > 0)
+            if bat.p_demand is not None:
+                for t in range(1, self.T):
+                    if bat.p_demand[t] > 0:
+                        # clip required charge to ax charging power if needed 
+                        # and leave some air to breathe for the optimizer
+                        p_demand = bat.p_demand[t]
+                        if p_demand >= bat.c_max * self.time_series.dt[t] / 3600. :
+                            p_demand = bat.c_max * self.time_series.dt[t] / 3600. * 0.9999
+                        self.problem += (self.variables['c'][i][t] >= p_demand)
+
             # Constraint (7): Minimum charge power limits
             if bat.c_min > 0:
```

Comment:
@ekkea ich hänge gerade noch hier. Besteht hier die Gefahr, dass wir uns ins Knie schießen weil das Modell "in Wirklichkeit" schon lädt, wir dann aber trotzdem nochmal versuchen, ihm `C_Min` abzuverlangen?

Ich würde bei Demand gerne die echte Ladeleistung angeben, das können ab Ende z.B. 2kW bei einem 3p Charger sein, mithin weniger als `C_Min`. Wenn es in einem Zeitschritt ein `P_Demand` gibt wäre es daher schön auf `C_Min` zu verzichten. Das liegt daran, dass ich beim Demand bereits lade und die Physik anders aussehen kann als das theoretische Limit. Da C_Min nicht pro Zeitschritt ist müsste diese Logik ins Modell?